### ISSUE TITLE
Polystrips: Source object is not updated after change the object 

### ISSUE THREAD

**Issue by OscarLeif** [CGCookie/retopoflow#285]
I still don't know if this is a bug, or usability.
Ok lets start.
When I want to make a retopology to an Object in the first time, polystrips doesn't have problems. So Ok 
I make my retopology and then I finish, so I want to make a quick retopology  to another object, well when you click the new object and make a new retopology polystrips the object source and object target will be still the previous object in order to fix this is very simple just delete the references in the Tool shelf Source and target.
I also made a quick video about showing the problem of usability ? I guess.
Basically it works fine but this make a little delays, in my case I create terrains in Blender by Sculpt then I retopology that by my self but polystrips can make a better distribution than I in the geometry workflow. Need to optimize geometry in my project.

Here in the video I explain that 
https://youtu.be/q9iWHAd0GDc

I think the same problem happen if you use PolyPen and contours.

**Comment by carter2422** [comment]
This is an interesting challenge @OscarLeif. The current behavior is intentional, but maybe it should be adapted. 

Currently the source/target objects have a background-process which runs constantly to check that the object still exists, and will clear the setting if the object is deleted. 

However, we really ought to find a better way of detecting if the artist is wanting to start a retopo-session on a new object, like in your use case.

I'll give this some thought and see what we can do. Potentially it may be as simple as switching the `source` to the active object if no source is set or if the active object is not the source.

**Comment by carter2422** [comment]
@vxlcoder any chance you could take a look at [the pull request for issue/218 merge](https://github.com/CGCookie/retopoflow/pull/287) to resolve the merge conflicts?

**Comment by carter2422** [comment]
I've opened an issue to resolve the core problem @OscarLeif. Closing this and will continue in #294
